plugins/woocommerce-admin/client/products/fields/attribute-field/add-attribute-modal.tsx

```diff
@@ -32,9 +35,15 @@ type AddAttributeModalProps = {
 };
 
 type AttributeForm = {
-	attributes: Array< HydratedAttributeType | { id: undefined; terms: [] } >;
+	attributes: Array< HydratedAttributeType | { id: undefined } >;
 };
 
+function isEmptyItem(
+	item: HydratedAttributeType | { id: undefined }
```

Comment:
Could this be a partial of `HydratedAttributeType` or does it need to have a separate `{ id: undefined }`?

Reply:
I wanted to have a clear differentiator between when a attribute was set and when it wasn't. Using a Partial of HydratedAttributeType would make that a bit more difficult and we did have to manually check if all the attributes exist on that property to deem it as the HydratedAttributeType in several places.

Reply:
I could potentially just try `null`, let me see if that works well.

Reply:
I updated it to using `HydratedAttributeType | null` so that things are more clear.

Reply:
Awesome!